### PR TITLE
fix(es/parser): syntax error for await in function

### PR DESCRIPTION
**Pull request by sosukesuzuki**
`await` isn't allowed in `function`

### CODE REVIEW

**Comment by sosukesuzuki** [comment]
Tests for `swc_ecma_transforms` did failed on CI. But I cannot reproduce failing tests on my local machine (Mac)... 

**terminal log:**

```
$ cargo test -p swc_ecma_transforms
Finished test [unoptimized + debuginfo] target(s) in 0.23s
     Running unittests (/Users/sosukesuzuki/ghq/github.com/sosukesuzuki/swc/target/debug/deps/swc_ecma_transforms-eb0b6f5f5e84c5ed)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests/decorators.rs (/Users/sosukesuzuki/ghq/github.com/sosukesuzuki/swc/target/debug/deps/decorators-020a530485df2235)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests/es2015_function_name.rs (/Users/sosukesuzuki/ghq/github.com/sosukesuzuki/swc/target/debug/deps/es2015_function_name-34d67af54256fe82)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

   Doc-tests swc_ecma_transforms

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

**Review by kdy1 — APPROVED** [review]
Thanks. I'll retrr CI.

**Comment by kdy1** [comment]
It turns out that one test is using `await` in a normal function. Can you remove the `await` from it?

**Comment by sosukesuzuki** [comment]
> It turns out that one test is using await in a normal function. Can you remove the await from it?

Thank you. I'll try it later.

**Comment by sosukesuzuki** [comment]
I've pushed commits to fix tests. Please check :pray:

File: ecmascript/parser/src/parser/expr/ops.rs [hunk 1]
```diff
@@ -322,7 +322,11 @@ impl<'a, I: Tokens> Parser<I> {
             })));
         }
 
-        if (self.ctx().in_async || self.syntax().top_level_await()) && is!(self, "await") {
+        if is!(self, "await") {
+            let ctx = self.ctx();
+            if ctx.in_function && !ctx.in_async {
+                syntax_error!(self, SyntaxError::AwaitInFunction);
```
Comment (kdy1) [inline ecmascript/parser/src/parser/expr/ops.rs:328]: I found that this error can be recovered, so we can report more issue in a single file.

File: ecmascript/parser/src/error.rs [hunk 2]
```diff
@@ -363,6 +365,8 @@ impl SyntaxError {
                 "for await syntax is valid only for for-of statement".into()
             }
 
+            SyntaxError::AwaitInFunction => "await isn't allowed in function".into(),
```
Comment (kdy1) [inline ecmascript/parser/src/error.rs:368]: Can you change this to `await isn't allowed in non-async function`?

**Comment by sosukesuzuki** [comment]
@kdy1 Thank you for reviewing! I've pushed commits addresses your reviews.

**Pull request merged**